src/main/client_context.cpp

```diff
@@ -95,17 +96,21 @@ unique_ptr<DataChunk> ClientContext::Fetch() {
 }
 
 string ClientContext::FinalizeQuery(ClientContextLock &lock, bool success) {
-	profiler.EndQuery();
+	profiler->EndQuery();
 	executor.Reset();
 
 	string error;
 	if (transaction.HasActiveTransaction()) {
 		ActiveTransaction().active_query = MAXIMUM_QUERY_ID;
-		query_profiler_history.GetPrevProfilers().emplace_back(transaction.ActiveTransaction().active_query,
-		                                                       move(profiler));
-		profiler.save_location = query_profiler_history.GetPrevProfilers().back().second.save_location;
-		if (query_profiler_history.GetPrevProfilers().size() >= query_profiler_history.GetPrevProfilersSize()) {
-			query_profiler_history.GetPrevProfilers().pop_front();
+		// Move the query profiler into the history
```

Comment:
The function `query_profiler_history->GetPrevProfilers()` is being called many times here in the span of a  few lines. This would look cleaner if you do this once:
```
auto &prev_profilers = query_profiler_history->GetPrevProfilers();
```

And calling e.g. `size()` and `pop_front()` on the reference.